@plotly/dash-test-components/src/components/ComponentAsProp.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+
+const ComponentAsProp = (props) => {
+    const { element, elements, id, shapeEl, list_of_shapes } = props;
+    console.log(list_of_shapes);
+    return (
+        <div id={id}>
+            {shapeEl && shapeEl.header}
+            {elements || element}
+            {shapeEl && shapeEl.footer}
+            {list_of_shapes && <ul>{list_of_shapes.map(e => <li key={e.value}>{e.label}</li>)}</ul> }
+        </div>
+    )
+}
+
+ComponentAsProp.propTypes = {
+    id: PropTypes.string,
+    element: PropTypes.node,
+
+    elements: PropTypes.arrayOf(PropTypes.node),
```

Comment:
`children` props - which just have type `PropTypes.node` - allow a list of components, and this is reflected in the docstring we use for node, "a list of or a singular dash component, string or number"

And as your example `{elements || element}` shows, in practical usage a node or an array of nodes generally (always?) work the same.

So if there were a way we could make it work the same way for other node props that would be the best. Otherwise we'll need to carve out a special case for `children` in the generated docs, vs other node props.

Reply:
Changed, now nodes have the same behavior in all cases.

Reply:
Can we then get rid of `elements` as a prop here, and instead in the tests let `element` and some of the nested props be lists of components?